Microsoft.HealthVault/Thing/ThingBase.cs

```diff
@@ -1237,7 +1237,7 @@ private void AddXmlSectionValues(XPathNavigator thingNavigator)
             // Check for the "data-xml" data-group
 
             XPathNodeIterator dataXmlIterator =
-                thingNavigator.Select("data-xml");
+                thingNavigator.Select("thing/data-xml");
```

Comment:
Yeah, you got an excellent point. I just did a different change to this. This one is less invasive and I think will fix the original problem. The issue we had is that tests are using all internal functions to test deserialization but we didn't test the public methods.